### PR TITLE
Date picker V2

### PR DESCRIPTION
**Pull request by dwarika-porter**

### CODE REVIEW

File: src/components/generic/form/form-input/FormDateInput.re [hunk 1]
```diff
@@ -1,5 +1,5 @@
 type t = {
-  value: Js.Date.t,
+  value: option(Js.Date.t),
```
Comment (ashutheone) [inline src/components/generic/form/form-input/FormDateInput.re:2]: Lets make a v2 component with a deadline for 15th April to replace v1 or to be reverted.

File: src/components/generic/date-time-picker/DatePicker.re [hunk 2]
```diff
@@ -1,9 +1,13 @@
+module OptionMomentToMomentMap = {
+  external optionMomentToMoment: Js.Null.t(MomentRe.Moment.t) => MomentRe.Moment.t = "%identity";
+};
+
```
Comment (ashutheone) [inline src/components/generic/date-time-picker/DatePicker.re:4]: Dont place this module here..
Maybe move this to DateTime utils or Option utils or something related.

**Review by ashutheone — CHANGES_REQUESTED** [review]

File: src/components/generic/form/form-input/FormDateInputV2.re [hunk 3]
```diff
@@ -0,0 +1,29 @@
+type t = {
+  value: option(Js.Date.t),
+  minDate: option(Js.Date.t),
+  maxDate: option(Js.Date.t),
+  onChange: Js.Date.t => unit,
+  format: string,
+};
+
+[@react.component]
+let make = (~input_props: FormInputProps.t('b, 'c), ~date_props: t) => {
```
Comment (ashutheone) [inline src/components/generic/form/form-input/FormDateInputV2.re:10]: Use make_props like [here](https://github.com/porterin/reason-utils/blob/master/src/components/generic/form/form-input/FormSelectInput.re)

File: src/components/generic/date-time-picker/DatePickerV2.re [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      value={Js.Null.fromOption(value) |> OptionMomentToMomentMap.optionMomentToMoment}
+      minDate
+      format
+      maxDate
+      disableFuture
+      disablePast
+      autoOk=true
+      disableToolbar=true
+      disabled
+      variant="inline"
```
Comment (ashutheone) [inline src/components/generic/date-time-picker/DatePickerV2.re:31]: pass this as prop and make it's default value inline in make_props

File: src/components/generic/date-time-picker/DatePickerV2.re [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      label
+      inputVariant="outlined"
```
Comment (ashutheone) [inline src/components/generic/date-time-picker/DatePickerV2.re:33]: For each value being set here.. make them default through make_props

**Review by ashutheone — CHANGES_REQUESTED** [review]

File: src/components/generic/form/form-input/FormDateInput.re (hunk 1 above)
Comment (dwarika-porter) [reply src/components/generic/form/form-input/FormDateInput.re]: Comment added

**Review by kd4001 — APPROVED** [review]

File: src/components/generic/date-time-picker/DatePickerV2.re [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      ~inputVariant: string,
+      ~autoOk: bool,
+      ~className: string
+    )
+    : React.element => {
+  <MuiPickersUtilsProvider utils=MuiPickersUtilsProvider.utils>
+    <MUIDatePicker
+      onChange
+      value={
+        Js.Null.fromOption(value)
```
Comment (Prateek13727) [inline src/components/generic/date-time-picker/DatePickerV2.re:26]: value -> Js.Null.fromOption -> OptionTypeUtils.OptionMomentToMomentMap.optionMomentToMoment

Comment (Prateek13727) [reply src/components/generic/date-time-picker/DatePickerV2.re:26]: more comprehensible

**Pull request merged**